### PR TITLE
PSK-16: endpoint to add new item

### PR DESCRIPTION
**Pull request by NerijusPuras**
Pushed interfaces into separate folder
Created POST request

### CODE REVIEW

File: WebAPI/Controllers/ItemController.cs [hunk 1]
```diff
@@ -20,4 +20,11 @@ public ItemController(IItemService itemService)
     {
         return await _itemService.GetItem(id);
     }
+
+    [HttpPost]
+    public async Task<IActionResult> CreateItem([FromBody] Item item)
```
Comment (somkamarius) [inline WebAPI/Controllers/ItemController.cs:25]: Open for discussion, but im thinking maybe its worth it to have a smaller dto for uploading an item? Perhaps FE could send in this kind of data and then the magic is done in the BE service itself:
`{
  "name": "string",
  "description": "string",
  "image": "string",
  "condition": 0,
  "category": 0,
  "uploadDate": "2022-04-11T07:20:11.242Z",
  "userId": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
  "addressId": "3fa85f64-5717-4562-b3fc-2c963f66afa6"
}`
Took out some imo trivial data and generated ids, left the essentials.

Comment (VeronikaSum) [reply WebAPI/Controllers/ItemController.cs:25]: If we will have editing functionality then sure we can have post (item) that is only partially built :)

**Review by VeronikaSum — APPROVED** [review]

File: WebAPI/Controllers/ItemController.cs (hunk 1 above)
Comment (NerijusPuras) [reply WebAPI/Controllers/ItemController.cs:25]: Yes, indeed. But your dto needs `From `and `To `dates additionality as it is provided by the user

File: WebAPI/Services/ItemService.cs [hunk 2]
```diff
@@ -3,14 +3,47 @@
...
+    {
+        Item item = buildItemEntity(partialItem);
+
+        await _context.Items.AddAsync(item);
+        await _context.SaveChangesAsync();
+
+        return item.ItemId;
+    }
+
+    private Item buildItemEntity(PartialItem partialItem)
```
[the remark this answers is not in the document]
Comment (VeronikaSum) [reply WebAPI/Services/ItemService.cs]: Fixed

**Pull request merged**